### PR TITLE
add role to user

### PR DESCRIPTION
**Pull request by NgengChhengkim**

### CODE REVIEW

File: app/views/devise/shared/_sign_up_form.html.erb [hunk 1]
```diff
@@ -29,6 +29,25 @@
       class: "form-control" %>
     <br />
 
+    <%= f.select :role, options_for_select(Settings.signup_roles.keys
```
Comment (yongsokheng) [inline app/views/devise/shared/_sign_up_form.html.erb]: `User.roles.keys.drop(1).map {|role| [t("devise.labels.#{role}"), role]}`
put in controller

File: app/views/devise/shared/_sign_up_form.html.erb [hunk 2]
```diff
@@ -29,6 +29,25 @@
...
+      .map{|role| [t("devise.labels.#{role}"), role]}, f.object.role),
+      {include_blank: t("devise.placeholders.select_role")}, {class: "form-control"} %>
+    <br />
+
+    <div class="gender">
+      <%= f.fields_for :user_profile, f.object.build_user_profile do |builder| %>
+
+        <%= builder.select :gender, options_for_select(UserProfile.genders.keys
+          .map{|gender| [t("devise.labels.genders.#{gender}"), gender]}),
```
Comment (yongsokheng) [inline app/views/devise/shared/_sign_up_form.html.erb:41]: put in controller

Comment (NgengChhengkim) [reply app/views/devise/shared/_sign_up_form.html.erb:41]: why?

File: config/application.yml [hunk 3]
```diff
@@ -1,5 +1,13 @@
 defaults: &defaults
 
+  signup_roles:
```
Comment (yongsokheng) [inline config/application.yml]: bo

File: app/views/devise/shared/_sign_up_form.html.erb (hunk 2 above)
Comment (yongsokheng) [reply app/views/devise/shared/_sign_up_form.html.erb:41]: logic thi viet trong controller di

Comment (NgengChhengkim) [reply app/views/devise/shared/_sign_up_form.html.erb:41]: khong phai phan logic. neu viet chi viet `UserProfile.genders.keys` thoi.

File: app/controllers/registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class RegistrationsController < Devise::RegistrationsController
+  def create
+    role = params[:user][:role]
+    if role == Settings.roles.admin
```
Comment (yongsokheng) [inline app/controllers/registrations_controller.rb]: viet mot dong co ve dep hon :D + sao ko dung enum

File: app/views/devise/shared/_sign_up_form.html.erb (hunk 2 above)
Comment (yongsokheng) [reply app/views/devise/shared/_sign_up_form.html.erb:41]: dung roi

File: app/controllers/registrations_controller.rb (hunk 4 above)
Comment (NgengChhengkim) [reply app/controllers/registrations_controller.rb]: phai viet super nua. viet mot dong khong duoc

File: app/views/devise/shared/_sign_up_form.html.erb (hunk 2 above)
Comment (NgengChhengkim) [reply app/views/devise/shared/_sign_up_form.html.erb:41]: nhung o day neu viet trong role thi phai them mot action new trong controller nua

Comment (yongsokheng) [reply app/views/devise/shared/_sign_up_form.html.erb:41]: ok, the cung dc. de trong controller di vay

File: app/controllers/registrations_controller.rb (hunk 4 above)
Comment (yongsokheng) [reply app/controllers/registrations_controller.rb]: dung ? : di

File: app/views/devise/shared/_sign_up_form.html.erb (hunk 2 above)
Comment (NgengChhengkim) [reply app/views/devise/shared/_sign_up_form.html.erb:41]: de o day co duoc khong vay? khong muon them trong controller. khong biet trong devise neu override co can them gi nua khong.

Comment (yongsokheng) [reply app/views/devise/shared/_sign_up_form.html.erb:41]: ok. :dancer:

**Review by yongsokheng — APPROVED** [review]

**Pull request merged**